data/src/main/java/com/example/pinterbest/data/repository/ProfileRepositoryImpl.kt

```diff
@@ -0,0 +1,37 @@
+package com.example.pinterbest.data.repository
+
+import android.content.SharedPreferences
+import com.example.pinterbest.data.api.ApiClient
+import com.example.pinterbest.data.models.toProfile
+import com.example.pinterbest.domain.common.ErrorMessage
+import com.example.pinterbest.domain.common.Result
+import com.example.pinterbest.domain.entities.Profile
+import com.example.pinterbest.domain.repositories.ProfileRepository
+import java.net.UnknownHostException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import retrofit2.HttpException
+
+class ProfileRepositoryImpl(val preferences: SharedPreferences) : ProfileRepository {
+    override suspend fun getProfileDetails(): Flow<Result<Profile>> = flow {
+        emit(Result.Loading)
+        try {
+            val profile = ApiClient().getInstance().getClient().getProfile(
+                SessionRepositoryImpl(preferences).authProvider() ?: ""
```

Comment:
SessionRepositoryImpl must be provided outside of class via constructor. Creating any dependencies inside it's a bad practice

Reply:
Fixed